src/core/core.helpers.js

```diff
@@ -544,8 +544,10 @@ module.exports = function(Chart) {
 		// If no style has been set on the canvas, the render size is used as display size,
 		// making the chart visually bigger, so let's enforce it to the "correct" values.
 		// See https://github.com/chartjs/Chart.js/issues/3575
-		canvas.style.height = height + 'px';
-		canvas.style.width = width + 'px';
+		if (!canvas.style.height && !canvas.style.width) {
```

Comment:
@andersponders shouldn't we split this check in case only one dimension is set in the style?

Reply:
You could, but I would say if the user is explicitly setting one through CSS, it seems more natural to let the other get handled that way too. im too lazy to make an example, but in my case I was setting height to be 70% of the container, and in that case i probably don't want the library to set width to be static 200px

Reply:
i should add this is with responsive=false, another option is instead check if responsive is set

Reply:
This method should not be aware about the responsive stuff, let's keep it the way you did.